streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableTransformValues.java

```diff
@@ -141,16 +144,32 @@ public void close() {
 
         @Override
         public void init(final ProcessorContext context) {
+            internalProcessorContext = (InternalProcessorContext) context;
             parentGetter.init(context);
             valueTransformer.init(new ForwardingDisabledProcessorContext(context));
         }
 
         @Override
         public ValueAndTimestamp<V1> get(final K key) {
             final ValueAndTimestamp<V> valueAndTimestamp = parentGetter.get(key);
-            return ValueAndTimestamp.make(
+
+            final ProcessorRecordContext currentContext = internalProcessorContext.recordContext();
+
+            internalProcessorContext.setRecordContext(new ProcessorRecordContext(
```

Comment:
Are there ever situations where users would want the old behavior (to have access to the `ProcessorContext` for the record that triggered the lookup, rather than the context for the record that's being looked up)? For example, if the topic name is relevant for the transformer and all records (including the current one that triggered the lookup and the one being processed) are from the same topic, then the old behavior gives access to the topic name but this new behavior doesn't.

Reply:
I don't think so, because when the write the transformer code, they won't even know that it might be executed during a lookup. When they write the transformer, they would assume they process a regular input record, and thus the transformer logic does not know anything about some other record that triggers the lookup.

Also, the record that triggers the lookup will be processed by its own processor and will have its own context, to access it's topic name.

Reply:
Thanks, all. I share the concern that this change could suddenly break something. Since the transformer has access to the ProcessorContext, people well may be accessing these members within the `transform` method. For example, a transformer may suddenly start to get an NPE on the topic name, and so forth.

This behavior change might be ok on the grounds that the context we previously provided was actually incorrect, though. If we do go ahead with this approach, we should be sure to make note of this potential in the upgrade notes. It wouldn't hurt to have a ticket as well, for visibility's sake.

Reply:
> This behavior change might be ok on the grounds that the context we previously provided was actually incorrect, though

Agreed. Also note, that the JavaDocs of `RecordContext` state, that `topic()` might return `null` if it's not available. Of course, user might not know why it's not available and it might be worth to point this out for this case. During regular processing, it's a sane assumption that the context information will always be available.

Reply:
Follow up PR: https://github.com/apache/kafka/pull/10731